test/persistence-hooks.suite.js

```diff
@@ -736,10 +739,18 @@ module.exports = function(dataSource, should) {
           { id: existingInstance.id, name: 'updated name' },
           function(err, instance) {
             if (err) return done(err);
-            observedContexts.should.eql(aTestModelCtx({
-              where: { id: existingInstance.id },
-              data: { id: existingInstance.id, name: 'updated name' }
-            }));
+            if (dataSource.connector.findOrCreate) {
+              observedContexts.should.eql(aTestModelCtx({
+                where: { id: existingInstance.id },
+                data: { id: existingInstance.id, name: 'updated name' }
```

Comment:
Actually, it's not a typo - whenever an optimized (dataSource.connector.findOrCreate) call can be made, the `currentInstance` will not be available. However, in the normal, unoptimized scenario, `updateAttributes` is invoked internally, which *does* set `currentInstance`. This is why I had to make a distinction here.